Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,224 @@
+            float weightInKg = PromptFloat("What is your weight in kg? ");
+            float heightInCm = PromptFloat("What is your height in cm? ");
+
+            Console.WriteLine($"{fullName} is {age} years old, his weight is {weightInKg} kg and his height is {heightInCm} cm.");
+            Console.WriteLine($"His BMI is {CalculateBmi(weightInKg, heightInCm / 100)}.");
+        }
+
+        /// <summary>
+        /// Prompts for an integer value. Converts the string input to an integer if possible.
+        /// </summary>
+        /// <param name="message">The message to print before the prompt.</param>
+        /// <param name="result">The parsing result. Is 0 if the parsing failed.</param>
+        /// <param name="input">The user input.</param>
+        /// <returns>Returns true if the parsing succeeded, false otherwise.</returns>
+        private static bool PromptTryParse(string message, out int result, out string input)
+        {
+            input = PromptInput(message);
+            return int.TryParse(input, out result);
+        }
+
+        /// <summary>
+        /// Prompts for a float value. Converts the string input to a float if possible.
+        /// </summary>
+        /// <param name="message">The message to print before the prompt.</param>
+        /// <param name="result">The parsing result. Is 0 if the parsing failed.</param>
+        /// <param name="input">The user input.</param>
+        /// <returns>Returns true if the parsing succeeded, false otherwise.</returns>
+        private static bool PromptTryParse(string message, out float result, out string input)
+        {
+            input = PromptInput(message);
+            return float.TryParse(input, NumberStyles.Float,
+                CultureInfo.InvariantCulture, out result);
+        }
+
+        /// <summary>
+        /// Prints the message and reads in the user input.
+        /// </summary>
+        /// <param name="message">The message to print.</param>
+        /// <returns>The unparsed and unvalidated input.</returns>
+        private static string PromptInput(string message)
+        {
+            Console.WriteLine(message);
+            return Console.ReadLine();
+        }
+
+        /// <summary>
+        /// Returns a value that indicates the type of input fault./>
+        /// </summary>
+        /// <param name="input">The input the user wrote.</param>
+        /// <returns>An integer indicating the error.</returns>
+        private static int InvalidInput(string input)
+        {
+            return string.IsNullOrEmpty(input)
+                ? EmptyNumberInput
+                : PrintInvalidNumber(input);
+        }
+
+        /// <summary>
+        /// Checks if the given value is in a valid range.
+        /// </summary>
+        /// <param name="value">The value to check.</param>
+        /// <returns>Returns the value if it is in range or a value indicating the opposite.</returns>
+        private static int RangeCheck(int value)
```

Comment:
This doesn't check, this adjusts a number too. Could be a better name.

Reply:
If you use advanced features like ternary operator, you might as well use generics.

Reply:
> If you use advanced features like ternary operator, you might as well use generics.

I agree that from the technical perspective generics would have been handy here. I thought that it would be closer to the current scope if I only use what Codacy suggests me (and possibly others) do change.

Reply:
My point was not using generics, but my point was not using ternary either, if we use only the tools that beginners have.

Reply:
I see.
Should I change it?

Reply:
I think so. It would put you on equal terms with students.